compat/mingw.c

```diff
@@ -1498,10 +1499,35 @@ struct pinfo_t {
 static struct pinfo_t *pinfo = NULL;
 CRITICAL_SECTION pinfo_cs;
 
+static void kill_child_processes_on_signal(void)
+{
+	DWORD status;
+
+	/*
+	 * Only continue if the process was terminated by a signal, as
+	 * indicated by the exit status (128 + sig_no).
+	 */
+	if (!GetExitCodeProcess(GetCurrentProcess(), &status) || status < 128)
```

Comment:
I'm not sure about this here, as `status` could be `STILL_ACTIVE` which has a value of `259`.

From MSDN:
> This function returns immediately. If the process has not terminated and the function succeeds, the status returned is STILL_ACTIVE.